Update documentation

ChrisGarty opened the pull request:
- moved Calling CDS info to mobileapp readme
- added key code examples for Calling CDS
- updated CDS callout in top level readme

ChrisGarty committed:
Update documentation
- move Calling CDS info to mobileapp readme
- add key code examples for Calling CDS
- updated CDS callout in top level readme

msftclas commented:
@ChrisGarty,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

ChrisGarty committed:
Update documentation
- add key code examples for Calling CDS

ChrisGarty committed:
Update solution to
- remove references to Nima's domain
- remove temporary OData code
- add comments to EmployeeController Get for use in readme
- add comments to CDSDataProvider.GetAllRecordsAsync for use in readme
